apps/daimo-mobile/src/view/screen/send/SendScreen.tsx

```diff
@@ -48,7 +51,34 @@ const SegmentedControlFixed = SegmentedControl as any;
 
 export default function SendScreen({ route }: Props) {
   console.log(`[SEND] rendering SendScreen ${JSON.stringify(route.params)}}`);
-  const { recipient, dollars, requestId } = route.params || {};
+  const { link } = route.params || {};
+  const linkStatus = useFetchLinkStatus(link);
+
+  const { recipient, dollars, requestId } = useMemo(() => {
+    if (
+      !linkStatus ||
+      linkStatus.isFetching ||
+      linkStatus.error ||
+      linkStatus.data == null
+    )
+      return {};
+    if (linkStatus.data.link.type === "account") {
+      const status = linkStatus.data as DaimoAccountStatus;
+      return {
+        recipient: status.account,
+        dollars: undefined,
+        requestId: undefined,
+      };
+    } else {
+      assert(linkStatus.data.link.type === "request");
+      const status = linkStatus.data as DaimoRequestStatus;
+      return {
+        recipient: status.recipient,
+        dollars: parseFloat(status.link.dollars),
+        requestId: status.requestId,
+      };
+    }
+  }, [linkStatus]);
```

Comment:
doesn't have to be in this PR, but in future we'll have to distinguish `recipientCanonical` from `recipientMatched` in search, see #214